schema/nested_attributes.go

```diff
@@ -59,6 +66,16 @@ func (s singleNestedAttributes) getNestingMode() nestingMode {
 	return nestingModeSingle
 }
 
+// ApplyTerraform5AttributePathStep applies the given AttributePathStep to the
+// nested attributes.
+func (s singleNestedAttributes) ApplyTerraform5AttributePathStep(step tftypes.AttributePathStep) (interface{}, error) {
+	if _, ok := step.(tftypes.ElementKeyString); !ok {
+		return nil, fmt.Errorf("cannot apply step %T to SingleNestedAttributes", step)
+	}
+
+	return s.nestedAttributes, nil
```

Comment:
I think we'd want to return _just_ the attribute named in the path step.

Reply:
Thinking about it, I think this method should just go on the `nestedAttributes` type, with `SingleNestedAttributes` inheriting it and the others overwriting it. This would allow the `nestedAttributes` the others are returning to be indexed into.